file.go

```diff
@@ -10,14 +14,104 @@ import (
 var (
 	_ fs.File     = (*file)(nil)
 	_ fs.FileInfo = (*fileInfo)(nil)
+	_ io.Seeker   = (*file)(nil)
 )
 
 type file struct {
-	io.ReadCloser
-	stat func() (fs.FileInfo, error)
+	cl     s3iface.S3API
+	bucket string
+	name   string
+
+	realReader      io.ReadCloser
```

Comment:
please revert and leave `io.ReadCloser` embedded - you don't need to implement `Close` then

Reply:
true. done